src/components/Dialog/index.js

```diff
@@ -28,13 +28,8 @@ export const Dialog = React.forwardRef(({
   children,
   ...props
 }, ref) => {
-  useEffect(() => {
```

Comment:
fixed in Prime https://github.com/primefaces/primereact/blob/master/CHANGELOG.md#1000-2023-09-28